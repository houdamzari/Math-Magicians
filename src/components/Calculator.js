import React, { useState } from 'react';
// import Buttons from './Buttons';
import calculate from '../logic/calculate';

const Calculator = () => {
  const details = { total: 0, next: null, operation: null };
  const [state, setState] = useState(details);

  const buttonClicked = (e) => {
    const buttonName = e.target.innerText;
    const result = calculate(state, buttonName);
    setState(result);
  };
  const { total, next, operation } = state;

  return (
    <>
      <div className="calculator__wrapper">
        <h2>Let`s do some maths</h2>
        <div className="calculator">
          <div className="results">
            {total}
            {operation}
            {next}
          </div>

          <div className="mathOperations">
            <button type="button" onClick={buttonClicked}>
              AC
            </button>
            <button type="button" onClick={buttonClicked}>
              +/-
            </button>
            <button type="button" onClick={buttonClicked}>
              %
            </button>
            <button
              type="button"
              className="mathsSymbol"
              onClick={buttonClicked}
            >
              &#247;
            </button>
            <button type="button" onClick={buttonClicked}>
              7
            </button>
            <button type="button" onClick={buttonClicked}>
              8
            </button>
            <button type="button" onClick={buttonClicked}>
              9
            </button>
            <button
              type="button"
              className="mathsSymbol"
              onClick={buttonClicked}
            >
              x
            </button>
            <button type="button" onClick={buttonClicked}>
              4
            </button>
            <button type="button" onClick={buttonClicked}>
              5
            </button>
            <button type="button" onClick={buttonClicked}>
              6
            </button>
            <button
              type="button"
              className="mathsSymbol"
              onClick={buttonClicked}
            >
              -
            </button>
            <button type="button" onClick={buttonClicked}>
              1
            </button>
            <button type="button" onClick={buttonClicked}>
              2
            </button>
            <button type="button" onClick={buttonClicked}>
              3
            </button>
            <button
              type="button"
              className="mathsSymbol"
              onClick={buttonClicked}
            >
              +
            </button>
            <button type="button" className="zero" onClick={buttonClicked}>
              0
            </button>
            <button type="button" onClick={buttonClicked}>
              .
            </button>
            <button
              type="button"
              className="mathsSymbol"
              onClick={buttonClicked}
            >
              =
            </button>
          </div>
        </div>
      </div>
    </>
  );
};

export default Calculator;
