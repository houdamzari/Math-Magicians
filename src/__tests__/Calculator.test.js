import React from 'react';
import renderer from 'react-test-renderer';

import Calculator from '../components/Calculator';

it('it matches snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
