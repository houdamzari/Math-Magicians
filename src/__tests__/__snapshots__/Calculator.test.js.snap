// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div
  className="calculator__wrapper"
>
  <h2>
    Let\`s do some maths
  </h2>
  <div
    className="calculator"
  >
    <div
      className="results"
    >
      0
    </div>
    <div
      className="mathOperations"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div
  className="calculator__wrapper"
>
  <h2>
    Let\`s do some maths
  </h2>
  <div
    className="calculator"
  >
    <div
      className="results"
    >
      0
    </div>
    <div
      className="mathOperations"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="mathsSymbol"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
